### PR TITLE
Query : Ban & UnBan node

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
- Added support for query `BanNode`
- Added support for query `UnBanNode`

## Changes
- Added support for query `BanNode`
- Added support for query `UnBanNode`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,11 +215,23 @@ SemVer getVersion()
...
+- `getNodeInfo`
 ```java
 NodeInfo getNodeInfo()
 ```
 
+- `banNode`
+```java
+boolean banNode(final BanNodeRequest request) throws Exception 
+```
+Ban a specific node.
```
Comment (EmilLa1) [inline README.md:227]: Please document what the `boolean` return value means.

File: README.md [hunk 2]
```diff
@@ -215,11 +215,23 @@ SemVer getVersion()
...
+
+- `unBanNode`
+```java
+boolean unBanNode(final InetAddress ip)
```
Comment (EmilLa1) [inline README.md:231]: Please document what the `boolean` return value means.

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 3]
```diff
@@ -325,6 +326,48 @@ public Semver getVersion() {
...
+                    .build();
+        } else {
+            throw new Exception("Invalid Request. Should mention either Ip or Id");
+        }
+
+        return server().banNode(peerElement).getValue();
+    }
+
+    /**
+     * Un Ban a specific node.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:358]:
```suggestion
     * Unban a specific node.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 4]
```diff
@@ -325,6 +326,48 @@ public Semver getVersion() {
...
+                    .setIp(StringValue.of(request.getIp().getHostAddress()))
+                    .build();
+
+        } else if (request.hasId()) {
+            peerElement = ConcordiumP2PRpc.PeerElement
+                    .newBuilder()
+                    .setNodeId(StringValue.of(request.getId()))
+                    .build();
+        } else {
+            throw new Exception("Invalid Request. Should mention either Ip or Id");
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:351]: I don't think this `Exception` is necessary as you are already enforcing that either the IP or the id is set in the request.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java:358]: done

File: README.md (hunk 1 above)
Comment (parv0888) [reply README.md:227]: done

File: README.md (hunk 2 above)
Comment (parv0888) [reply README.md:231]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java:351]: removed

**Review by EmilLa1 — APPROVED** [review]

**Pull request merged**